src/main/java/decoder/Decoder.java

```diff
@@ -0,0 +1,34 @@
+package decoder;
+
+import com.google.protobuf.util.JsonFormat;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import proto.SensorData.Pose;
+
+/** Decode data from the proto format to JSON. */
+public class Decoder {
```

Comment:
You can make it a generic class.

Reply:
I think this is a good point, but would likely sense to implement in a later pull request. For MVP I think the current implementation should be sufficient.